Feature/camera movement/#86bzqvu6j

MoisesGarcia1 opened the pull request:
Updated Cinematic Camera

MoisesGarcia1 committed:
Added Cinematic Camera

Matherthebec committed:
Partial fix for CinematicCamera

Back to Main Camera, but getting gizmo and axis from editorcamera

MoisesGarcia1 committed:
Fixed issue returning to main camera after a cinematic

MoisesGarcia1 committed:
Merge branch 'origin/gameplay/camera-movement' of https://github.com/RollingBarrel/Helheim-Engine into origin/gameplay/camera-movement

MoisesGarcia1 committed:
Merge branch 'develop' into origin/gameplay/camera-movement

MoisesGarcia1 committed:
Last develop version update

MoisesGarcia1 committed:
Added cinematic filter

MoisesGarcia1 committed:
Updated cinematic camera system with different cameras

MoisesGarcia1 committed:
Deleted unused code

MoisesGarcia1 committed:
Added conditionals to avoid gameobject crashes with null gameobjects

MoisesGarcia1 committed:
Fixed frame issue chaging between main camera and cinematic camera

MoisesGarcia1 committed:
Level1Cinematic1 and Level1Cinematic2 scene for testing

MoisesGarcia1 committed:
Update Level1_Cinematic1.scn

MoisesGarcia1 committed:
Dummy animations first test

MoisesGarcia1 committed:
Dummy animation second WIP version

MoisesGarcia1 committed:
Test Dummy Cinematics

MoisesGarcia1 committed:
Deleted commented code

MoisesGarcia1 committed:
Imrpoved Activate Camera Call

MoisesGarcia1 committed:
Update InitAnimation function

MoisesGarcia1 committed:
Update Level1Cinematic2

MoisesGarcia1 committed:
Merge branch 'develop' into origin/gameplay/camera-movement

MoisesGarcia1 committed:
Update Level1_Cinematic.scn

MoisesGarcia1 committed:
Level2_CinematicCamera

MoisesGarcia1 committed:
Added mCameraPosition to PlayerCamera

MoisesGarcia1 committed:
Delete Level1_Cinematic2.scn

MoisesGarcia1 committed:
Changed local Delay function by TimerScript Delay function

MoisesGarcia1 committed:
Fade() - First version

MoisesGarcia1 committed:
Fade() Version 2

MoisesGarcia1 committed:
Fade on/off working

MoisesGarcia1 committed:
Fix to locate the player in a correct position behind the doors in level 1

MoisesGarcia1 committed:
Added new transition of fade on / off

MoisesGarcia1 committed:
Squashed commit of the following:

commit 6e9aae41c6dcf807eb1bf14960b7c1f81ee85ba8
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Wed Aug 28 13:46:23 2024 +0200

    Added new transition of fade on / off

commit 14bda6f9570b26418c080440c4ad3526cf726ced
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Sun Aug 25 14:00:47 2024 +0200

    Fix to locate the player in a correct position behind the doors in level 1

commit 09b5db37271b488807fe95214c18022402dbd9f8
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Sun Aug 25 12:33:20 2024 +0200

    Fade on/off working

commit b2b7a29ec730b2166de349974158d3c26faa4372
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Thu Aug 22 18:03:08 2024 +0200

    Fade() Version 2

commit d28efa5505c293a43a0b5196b6edf042291a0765
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Wed Aug 21 23:59:11 2024 +0200

    Fade() - First version

commit 718dda31511a824c89d1c089136a28cc23d5cb04
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Aug 19 15:33:35 2024 +0200

    Changed local Delay function by TimerScript Delay function

commit bff2f42af57fd533d2b45c1c446f11e544a03b97
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Aug 19 13:11:33 2024 +0200

    Delete Level1_Cinematic2.scn

commit 43c3deb827a37406f76f114cad65e2894ee8f60e
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Aug 19 00:01:24 2024 +0200

    Added mCameraPosition to PlayerCamera

commit 8557ce7a91c92b022469079ec2e974b4ce902aff
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Sun Aug 18 21:06:01 2024 +0200

    Level2_CinematicCamera

commit 369586576c3cc66695affe7e355130e7c96a33d7
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Sat Aug 17 22:09:50 2024 +0200

    Update Level1_Cinematic.scn

commit 900f517fe291c4338caf88c82464249a10dac05b
Merge: a2c4656a 07772fbf
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Fri Aug 16 17:21:22 2024 +0200

    Merge branch 'develop' into origin/gameplay/camera-movement

commit a2c4656add4f9e97202fafa3c6f708197ae60c51
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Fri Aug 16 17:13:39 2024 +0200

    Update Level1Cinematic2

commit 60bbe201ba56dd3a7723992d56f08777a71ddf0c
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Fri Aug 16 17:07:46 2024 +0200

    Update InitAnimation function

commit 6ac8bcb0ad3efa585184695f58748ea66569b8bb
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Thu Aug 15 11:52:29 2024 +0200

    Imrpoved Activate Camera Call

commit a7c51d31f1d4a437f3cb035d15d8c9b9facbae36
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Wed Aug 14 21:16:24 2024 +0200

    Deleted commented code

commit b6f97e3e6a87aee859111951a2372dbe683f98df
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Wed Aug 14 18:36:55 2024 +0200

    Test Dummy Cinematics

commit 2b4b7c9c88e71fb79b83a48d648c1ef7849dac1a
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Tue Aug 13 23:36:24 2024 +0200

    Dummy animation second WIP version

commit 526e28f0e3ccd3bfa72454f2ec0eaa7e8f585e1f
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Tue Aug 13 15:52:38 2024 +0200

    Dummy animations first test

commit a343a43f26b3b5b5d212c560e39eb02c81f1bd6a
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Aug 12 23:41:34 2024 +0200

    Update Level1_Cinematic1.scn

commit 1b3f7af199f035b9edd546d76b26c1d79fe67991
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Aug 12 23:18:16 2024 +0200

    Level1Cinematic1 and Level1Cinematic2 scene for testing

commit 09589a47506be09f54da766bee50b17de0456ed2
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Aug 12 22:30:57 2024 +0200

    Fixed frame issue chaging between main camera and cinematic camera

commit 65c6c92b965e95b88109996109a69cb685f30a48
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Aug 12 16:29:03 2024 +0200

    Added conditionals to avoid gameobject crashes with null gameobjects

commit ef98da909db685da1c16b75542acde9e72feb5b6
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Aug 12 13:01:13 2024 +0200

    Deleted unused code

commit e4a2ac1229e6e232dc1b22c490e39d25fb1ba764
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Aug 12 00:19:53 2024 +0200

    Updated cinematic camera system with different cameras

commit 3a5146bb1c68416a8d37c0a37d1711a20e4882c8
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Sun Aug 11 20:47:03 2024 +0200

    Added cinematic filter

commit 97a6f262512c5f497790dd5a4bccef09b213b83d
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Sun Aug 11 14:26:34 2024 +0200

    Last develop version update

commit 8d71fd30fdc9fb822efa2326c84b2197cc317f72
Merge: 8fecbeaf 65a2a190
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Sun Aug 11 12:46:11 2024 +0200

    Merge branch 'develop' into origin/gameplay/camera-movement

commit 8fecbeaf9317144bb377e97b83086f982a459151
Merge: c3000e29 3c2c6622
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Jul 29 15:26:19 2024 +0200

    Merge branch 'origin/gameplay/camera-movement' of https://github.com/RollingBarrel/Helheim-Engine into origin/gameplay/camera-movement

commit c3000e29d3d5c1b9987a5e5949fc405f3e359901
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Jul 29 15:23:23 2024 +0200

    Fixed issue returning to main camera after a cinematic

commit 3c2c6622dd204365e1d0e03c6ae73e0c848a9c37
Author: Giovanni Matera <matherthebec@gmail.com>
Date:   Mon Jul 29 07:05:16 2024 +0200

    Partial fix for CinematicCamera

    Back to Main Camera, but getting gizmo and axis from editorcamera

commit 5429c4b8b1f53fc93f73d39792b15a9a38ca162f
Author: Moises Garcia <39276654+MoisesGarcia1@users.noreply.github.com>
Date:   Mon Jul 29 00:43:21 2024 +0200

    Added Cinematic Camera

MoisesGarcia1 committed:
Update

MoisesGarcia1 committed:
Unused code commented

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#clickupid

MoisesGarcia1 committed:
Updated Level1Cinematic

MoisesGarcia1 committed:
Level2Cinematic scene

MoisesGarcia1 committed:
Skip button for Camera Cinematic

MoisesGarcia1 committed:
Cinematic Skip Button with fade off

MoisesGarcia1 committed:
Cinematic Skip Button Fade Off Fixed

MoisesGarcia1 committed:
Skipe Button bug fix on main camera

Fixed a crash when player uses skip button at the beginning of fade off when the main camera is still active.

MoisesGarcia1 committed:
CInematic Delay Time Bug Fix using Skipe Button

MoisesGarcia1 committed:
Added Start Game Controller Skip Button

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#clickupid

MoisesGarcia1 committed:
Level1Cinematic testing scene

MoisesGarcia1 committed:
Level2_Cinematic testing scene

MoisesGarcia1 committed:
Updated "B" button for gamepad and keyboard to skip cinematics

MoisesGarcia1 committed:
Updated Skip Keys

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Modified MainCamera const_cast

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

upcgames2324 commented:
Task linked: [CU-86bzqvu6j Enemies CutScenes](https://app.clickup.com/t/86bzqvu6j)

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Update Project

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
WIP

MoisesGarcia1 committed:
Changed custom Lerp function to use Float3 Lerp Function

MoisesGarcia1 committed:
Created UpdateCinematic

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Divided StartCinematic in some functions

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Updated UpdateCinematic and related functions

The cinematic camera is now working correctly, we just need to make sure that enemies are not spammed during cinematics.

MoisesGarcia1 committed:
BattleAreas now not respawn enemies during cinematics

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
WIP Prefabs

MoisesGarcia1 committed:
Modified method in Battlearea to activate enemies

MoisesGarcia1 committed:
Improvements showing dummies

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
WIP Cinematic Camera Rotation

MoisesGarcia1 committed:
Added Pause in CinematicCamera

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
WIP CinematicCamera

MoisesGarcia1 committed:
WIP

MoisesGarcia1 committed:
WIP

MoisesGarcia1 committed:
WIP

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Cinematic Camera improvements and Level1 and 2 prefabs

MoisesGarcia1 committed:
Level 1 and 2 cinematic prefabs

MoisesGarcia1 committed:
Patch

Patch that restores the screen to avoid a bug that appears when the player keys B when the fade is completely black

MoisesGarcia1 committed:
Level  3 Prefab

MoisesGarcia1 committed:
Deleted cinematic test scenes and restored original  scenes

MoisesGarcia1 committed:
Refactor GetInstance  in CinematicCamera

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Cinematic Camera black Screen bugfix

MoisesGarcia1 committed:
Improvements in cinematic bug fix

MoisesGarcia1 committed:
Restored Develop maps

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Solved merge bug

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Updated level scenes with cinematic cameras

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Updated levels with cinematic cameras

MoisesGarcia1 committed:
Added returns to HandleFadeOut and Fade

MoisesGarcia1 committed:
Merge branch 'develop' into feature/camera-movement/#86bzqvu6j

MoisesGarcia1 committed:
Changed Dummy enemytags
